src/pal/src/file/directory.cpp

```diff
@@ -255,9 +263,13 @@ RemoveDirectoryW(
         dwLastError = ERROR_PATH_NOT_FOUND;
         goto done;
     }
-    
-    mb_size = WideCharToMultiByte( CP_ACP, 0, lpPathName, -1, mb_dir, MAX_LONGPATH,
+
+    length = (PAL_wcslen(lpPathName)+1) * sizeof(WCHAR);
```

Comment:
Shouldn't we multiply by 4 if WCHAR >= 0x10000  requires 4 bytes (encoded as 2 WCHARs = 2*sizeof(WCHAR) = 4)